astropy/coordinates/solar_system.py

```diff
@@ -42,21 +45,39 @@
                                       )
 SOLAR_SYSTEM_BODIES = tuple(BODY_NAME_TO_KERNEL_SPEC.keys())
 
+_JPL_EPHEM_NOTE = """
+    This calculation uses JPL Ephemeris SPK filesto calculate the body's
+    location, defaulting to DE430.  It will be downloaded the first time this
+    function is used and cached from then on.  To change this, set
+    ``kernel_url`` as described in the coordinate documentation.
+"""[1:-1]
 
-def _download_spk_file(url=('http://naif.jpl.nasa.gov/pub/naif/'
-                            'generic_kernels/spk/planets/de430.bsp'),
-                       show_progress=True):
+class kernel_url(ScienceState):
     """
-    Get the Satellite Planet Kernel (SPK) file from NASA JPL.
+    The URL to use for downloading a download the Satellite Planet
+    Kernel (SPK) file with ephemerides.  The download will *not* occur when
+    this state is set, but rather whenever the first time is that the
+    kernel actually needs to be used.
 
-    Download the file from the JPL webpage once and subsequently access a
-    cached copy. The default is the file de430.bsp.
-
-    This file is ~120 MB, and covers years ~1550-2650 CE [1]_.
+    Notes
+    -----
+    The default Satellite Planet Kernel (SPK) file from NASA JPL (DE430) is
+    ~120MB, and covers years ~1550-2650 CE [1]_.
 
     .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
     """
-    return download_file(url, cache=True, show_progress=show_progress)
+    _value = ('http://naif.jpl.nasa.gov/pub/naif/generic_kernels'
+               '/spk/planets/de430.bsp')
+
+    @classmethod
+    def validate(cls, value):
+        try:
```

Comment:
I don't think so,  because you never know what random SPK file someone in the future might want to use, and we'd have to make assumptions about what's a "valid" name (or actually download the file and check it somehow, which is also non-ideal).  I think it's better to just validate here that it's a thing that might be downloadable, and let the `jplephem` and `download_file` machinery be responsible for the rest.  Otherwise it'll be a nightmare to maintain.